src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+        /// representing the DAB engine's supported HTTP verbs and relevant route restrictions:
+        /// Paths including primary key:
+        /// - GET (by ID), PUT, PATCH, DELETE
+        /// Paths excluding primary key:
+        /// - GET (all), POST
+        /// </summary>
+        /// <example>
+        /// A path with primary key where the parameter in curly braces {} represents the preceding primary key's value.
+        /// "/EntityName/primaryKeyName/{primaryKeyValue}"
+        /// A path with no primary key nor parameter representing the primary key value:
+        /// "/EntityName"
+        /// </example>
+        /// <returns>All possible paths in the DAB engine's REST API endpoint.</returns>
+        private OpenApiPaths BuildPaths()
+        {
+            OpenApiPaths pathsCollection = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                // Routes including primary key
+                Tuple<string, OpenApiPathItem> path = BuildPath(entityName, includePrimaryKeyPathComponent: false);
+                pathsCollection.Add(path.Item1, path.Item2);
+
+                // Routes excluding primary key
+                Tuple<string, OpenApiPathItem> pathGetAllPost = BuildPath(entityName, includePrimaryKeyPathComponent: true);
+                pathsCollection.TryAdd(pathGetAllPost.Item1, pathGetAllPost.Item2);
+            }
+
+            return pathsCollection;
+        }
+
+        /// <summary>
+        /// Includes Path with Operations(+responses) and Parameters
+        /// Parameters are the placeholders for pk values in curly braces { } in the URL route
+        /// localhost:5000/api/Entity/pk1/{pk1}/pk2/{pk2}
+        /// more generically:
+        /// /entityName OR /RestPathName followed by (/pk/{pkValue}) * Number of primary keys.
+        /// </summary>
+        /// <returns>Tuple of the path value: e.g. "/Entity/pk1/{pk1}"
+        /// and path metadata.</returns>
+        private Tuple<string, OpenApiPathItem> BuildPath(string entityName, bool includePrimaryKeyPathComponent)
+        {
+            SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+
+            string entityRestPath = GetEntityRestPath(entityName);
+            string entityBasePathComponent = $"/{entityRestPath}";
```

Comment:
shouldn't this path be prefixed by the global rest path prefix? e.g. the default one is /api/ - how does that get added?

Reply:
nvm saw it in the CreateDocument function: restEndpointPath